# DP: Description: Fix __bind redefinition problem
# DP: Related bugs: 
# DP: Dpatch author: Clint Adams
# DP: Patch author: Clint Adams
# DP: Upstream status: Not Submitted
# DP: Status Details:
# DP: Date: 2005-12-10

--- sysdeps/unix/sysv/linux/sparc/sparc32/socket.S.old	2003-08-31 13:23:11.000000000 -0400
+++ sysdeps/unix/sysv/linux/sparc/sparc32/socket.S	2005-12-10 12:46:23.313930095 -0500
@@ -40,7 +40,11 @@
    The .S files for the other calls just #define socket and #include this.  */
 
 #ifndef __socket
-#define __socket P(__,socket)
+# ifndef NO_WEAK_ALIAS
+#  define __socket P(__,socket)
+# else
+#  define __socket socket
+# endif
 #endif
 
 .globl __socket
@@ -105,4 +109,6 @@
 
 END (__socket)
 
+#ifndef NO_WEAK_ALIAS
 weak_alias (__socket, socket)
+#endif
--- sysdeps/unix/sysv/linux/sparc/sparc64/socket.S.old	2003-08-31 13:23:11.000000000 -0400
+++ sysdeps/unix/sysv/linux/sparc/sparc64/socket.S	2005-12-10 12:46:23.313930095 -0500
@@ -40,7 +40,11 @@
    The .S files for the other calls just #define socket and #include this.  */
 
 #ifndef __socket
-#define __socket P(__,socket)
+# ifndef NO_WEAK_ALIAS
+#  define __socket P(__,socket)
+# else
+#  define __socket socket
+# endif
 #endif
 
 .globl __socket
@@ -105,4 +109,6 @@
 
 END (__socket)
 
+#ifndef NO_WEAK_ALIAS
 weak_alias (__socket, socket)
+#endif
